snake_state.py

```diff
@@ -234,6 +241,9 @@ def _grow_step(self, old_tail_pos):
 
     def _respawn_apple(self):
         self.__apple.move(self.__grid.find_free_cell())
+        
+    def _get_dist_to_apple(self, head_pos):
+        return math.sqrt((head_pos[0] - self.apple_position[0])**2 + (head_pos[1] - self.apple_position[1])**2)
```

Comment:
Pythagorean theorem doesn't help us much here, we need the amount of squares to move, not the size of the vector pointing from head to apple.
Make it `dx + dy` instead.